### PR TITLE
add test to check type of df column after loc()

### LINKED ISSUE #23326: Data type of new assigned DataFrame column
```python

df = pd.DataFrame([list('abc')])
print(df)

df.loc[:, 'col1'] = int(5)
df['col2']        = int(5)
df                = df.assign(col3=int(5))

print(df.dtypes)
print(df)


  0  1  2
0  a  b  c
0        object
1        object
2        object
col1    float64
col2      int64
col3      int64
dtype: object
   0  1  2  col1  col2  col3
0  a  b  c   5.0     5     5
```
#### Problem description

The fact that `df.loc[:,'col1] = int(5)` yields a new column with dtype `float64` seems strange.

#### Expected Output

All three methods should yield a new column with dtype `int64`

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Linux
OS-release: 4.15.0-1023-aws
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: C.UTF-8
LOCALE: None.None

pandas: 0.23.4
pytest: 3.8.2
pip: 18.1
setuptools: 40.4.3
Cython: None
numpy: 1.15.0
scipy: 1.1.0
pyarrow: None
xarray: None
IPython: 5.8.0
sphinx: None
patsy: None
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.3
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
sqlalchemy: None
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by jackgoldsmith4**
- [x] closes #23326
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

**Comment by jackgoldsmith4** [comment]
Hey @mroeschke, I'd like to start writing tests for a lot of the existing GH issues. I am learning as I go, so I am not sure if something like this is sufficient and/or in the right place. Any critical feedback would be greatly appreciated so I can start writing better and more complex tests. Thanks!

**Comment by mroeschke** [comment]
Awesome! Definitely check out https://pandas.pydata.org/docs/dev/development/contributing_codebase.html#using-pytest for all the details of how tests ideally should be structured and what should be included

File: pandas/tests/indexing/test_loc.py [hunk 1]
```diff
@@ -2869,6 +2869,13 @@ def test_loc_setitem_using_datetimelike_str_as_index(fill_val, exp_dtype):
     tm.assert_index_equal(df.index, expected_index, exact=True)
 
 
+def test_loc_set_int_dtype():
+    df = pd.DataFrame([list("abc")])
+    df.loc[:, "col1"] = int(5)
```
Comment (mroeschke) [inline pandas/tests/indexing/test_loc.py:2874]:
```suggestion
    df.loc[:, "col1"] = 5
```

File: pandas/tests/indexing/test_loc.py [hunk 2]
```diff
@@ -2869,6 +2869,13 @@ def test_loc_setitem_using_datetimelike_str_as_index(fill_val, exp_dtype):
...
+
+    assert str(df.dtypes["col1"]) == "int64"
```
Comment (mroeschke) [inline pandas/tests/indexing/test_loc.py:2876]: Generally we want to compare the complete `DataFrame` result which will check the dtype e.g.

```
df.loc[:, "col1"] = 5
expected = DataFrame(...)
tm.assert_frame_equal(df, expected)
```

File: pandas/tests/indexing/test_loc.py [hunk 3]
```diff
@@ -2869,6 +2869,13 @@ def test_loc_setitem_using_datetimelike_str_as_index(fill_val, exp_dtype):
     tm.assert_index_equal(df.index, expected_index, exact=True)
 
 
+def test_loc_set_int_dtype():
```
Comment (mroeschke) [inline pandas/tests/indexing/test_loc.py:2872]: Add a comment below with the GH issue that this is fixing e.g. `# GH 123`

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jackgoldsmith4

**Pull request merged**